### PR TITLE
Methodology updates from meeting 17/11/2021

### LINKED ISSUE #9: For proper analysis of an android application a testing environment has to be configured.

### LINKED ISSUE #10: The research required that five applications were examined > The research required that each team meber has one application to examine.

### LINKED ISSUE #11: Behavior analysis: add keyword end user perspective.

### LINKED ISSUE #12: tools meer uitegebreid uitleggen network analysis.

### LINKED ISSUE #14: Countermeasures staan los van yara rule sets. hoe kan een gebruiker zich bewapenen tegen de applicaties.

### PR DESCRIPTION
**Pull request by Wolkje**

### CODE REVIEW

File: Verslag/methodology.tex [hunk 1]
```diff
@@ -1,32 +1,71 @@
...
 
 \subsection{Finding the malicious applications}
 
-For the research it was decided that all the selected applications had come from the site Koodous and were confirmed to be malicious with the site VirusTotal. This had been done individually by all team members to find applicable APK files for the research. Some checks were performed manually to see if the applications contained any pointers that indicated that they contained malware. The application was checked on its availability for download on the Google Play store. The potentially malicious application was compared with the variant from the Google Play store. A few examples of these pointers were differing sizes or if both had the exact same package name as the Google Play store variant while differing in permissions or certificates. If it had one of those pointers, then it likely was an indicator that there was something extra hidden in the code.
+For the research it was decided that all the selected applications had come from the site Koodous and were confirmed to be malicious with the site VirusTotal.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: had to come from the site Koodous and had to be confirmed

File: Verslag/methodology.tex [hunk 2]
```diff
@@ -1,32 +1,71 @@
...
+The application was checked on its availability for download on the Google Play store.
+The potentially malicious application was compared with the variant from the Google Play store.
+A few examples of these pointers were differing sizes or if both had the exact same package name as the Google Play store variant while differing in permissions or certificates.
+If it had one of those pointers, then it likely was an indicator that there was something extra hidden in the code.
 
-The application was retrieved from both Koodous and the Google Play Store. The APK from Google Play was downloaded using the apkcombo.com tool. Both APK files were uploaded to VirusTotal to check for any differences in permissions between the two versions of the application. Anything VirusTotal had marked anything as malicious, was noted. This was up to the discretion of every individual team member to decide if they decided to work on the chosen application for the duration of the project.
+The application was retrieved from both Koodous and the Google Play Store.
+The APK from Google Play was downloaded using the apkcombo.com tool.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: website? instead of tool

File: Verslag/methodology.tex [hunk 3]
```diff
@@ -1,32 +1,71 @@
...
+Both APK files were uploaded to VirusTotal to check for any differences in permissions between the two versions of the application.
+Anything VirusTotal had marked anything as malicious, was noted.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: 2x Anything. I would remove last anything

File: Verslag/methodology.tex [hunk 4]
```diff
@@ -1,32 +1,71 @@
...
+This was up to the discretion of every individual team member to decide if they decided to work on the chosen application for the duration of the project.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: It was up to the discretion of each individual team member to decide on the application they wanted to work on for the duration of the project

File: Verslag/methodology.tex [hunk 5]
```diff
@@ -1,32 +1,71 @@
...
 
 \subsection{Behavior analysis}
 
-The research required that five applications were examined. This had been done on a virtual computer with android studio installed or locally. During the installation of the application the network traffic had been captured. This had been done using Wireshark and mitmproxy. The network traffic was not relevant for behavior analysis but was captured as part of the network analysis.
+The research required that each team member has one application to examine.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: has > had

File: Verslag/methodology.tex [hunk 6]
```diff
@@ -1,32 +1,71 @@
...
+This had been done on a virtual computer with android studio installed or locally.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: This had been done on a virtual machine, with android studio installed, or locally.

File: Verslag/methodology.tex [hunk 7]
```diff
@@ -1,32 +1,71 @@
...
+During the installation of the application the network traffic had been captured.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: was captured

File: Verslag/methodology.tex [hunk 8]
```diff
@@ -1,32 +1,71 @@
...
+This had been done using Wireshark and mitmproxy.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: was done

File: Verslag/methodology.tex [hunk 9]
```diff
@@ -1,32 +1,71 @@
...
+The network traffic was not relevant for behavior analysis but was captured as part of the network analysis.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: for the behavior analysis

File: Verslag/methodology.tex [hunk 10]
```diff
@@ -1,32 +1,71 @@
...
 
-The behavior research had been predominantly about using the application and noting anything out of the ordinary. This had also been compared step by step to the original (non-malicious) application. The differences between the two are noted for further investigative purposes. The ordinary behavior had been investigated further during network and code analysis.
+The behavior research had been predominantly about using the application from a users point of view and noting anything out of the ordinary.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: This is a weird sentence, I don't know what you're trying to say. Might be The behavior analysis?

File: Verslag/methodology.tex [hunk 11]
```diff
@@ -1,32 +1,71 @@
...
+This had also been compared step by step to the original (non-malicious) application.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: had also been compared > this was also compared

File: Verslag/methodology.tex [hunk 12]
```diff
@@ -1,32 +1,71 @@
...
+The differences between the two are noted for further investigative purposes.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: are > were

File: Verslag/methodology.tex [hunk 13]
```diff
@@ -1,32 +1,71 @@
...
+The ordinary behavior had been investigated further during network and code analysis.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: had been > was

**Review by rafaeltab — CHANGES_REQUESTED** [review]
More comments coming

File: Verslag/methodology.tex [hunk 14]
```diff
@@ -1,32 +1,71 @@
...
 
 \subsection{Network analysis}
 
-During the behavior analysis the network traffic from the application was captured using Wireshark and mitmproxy. In the network analysis this captured network traffic had been analyzed using a combination of Wireshark, mitmproxy and shodan.io. During the analysis anything that stood out like any IP addresses and or domain names were written down.
+During the behavior analysis the network traffic from the application was captured using Wireshark and mitmproxy.
+In the network analysis this captured network traffic had been analyzed using a combination of Wireshark, mitmproxy and shodan.io.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: During the network analysis? idk bout this

had been analyzed > was analyzed

File: Verslag/methodology.tex [hunk 15]
```diff
@@ -1,32 +1,71 @@
...
+Wireshark was used to capture all network data that has been sent by the application and gave the opertunity to perform a deep analysis what exactly was sent.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: data that has been sent by the > data that was sent by the

File: Verslag/methodology.tex [hunk 16]
```diff
@@ -1,32 +1,71 @@
...
+Mitmproxy was used to capture all http and https requests, this was used to find what IP-addresses a connection was made to.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: Connections were made to

File: Verslag/methodology.tex [hunk 17]
```diff
@@ -1,32 +1,71 @@
...
+These applications turned the single .apk files into structured files, which were readable for the code analysis.
+This created the opportunity to read the code.
+An attempt was made to analyze the code.
+Analyzing the code was very difficult due to obfuscation.
+Where possible it was documented whether the code was malicious or not.
 
 \subsection{Process analysis}
-The Processes were analyzed to gain the insights that the research needs. The application had been started in the emulator of Android Studio. During the startup and use of the application, the Memory profiler had been used to monitor the memory usage of the application. All memory usage was documented and all peculiar behavior had been investigated further, to understand what had caused the suspicious behavior. Optionally, a heap dump was done during execution, which led to interesting findings.
+The Processes were analyzed to gain the insights that the research needs.
+The application had been started in the emulator of Android Studio.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: had been > was

File: Verslag/methodology.tex [hunk 18]
```diff
@@ -1,32 +1,71 @@
...
+During the startup and use of the application, the Memory profiler had been used to monitor the memory usage of the application.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: had been > was

File: Verslag/methodology.tex [hunk 19]
```diff
@@ -1,32 +1,71 @@
...
+All memory usage was documented and all peculiar behavior had been investigated further, to understand what had caused the suspicious behavior.
+Optionally, a heap dump was done during execution, which led to interesting findings.
 
 \subsection{Countermeasures and detection}
 
-To finalize the analysis, a YARA ruleset had been created for the application. All findings were documented into a report which was delivered to the team leader. These steps were performed by all the members. If an application needed additional steps this was declared at the start of the application chapter.
\ No newline at end of file
+To finalize the analysis, a YARA ruleset had been created for the application.
+And ways the user of the application can mitigate the effect of the malicious code.
```
Comment (rafaeltab) [inline Verslag/methodology.tex]: Also, methods to mitigate the effect of the malicious code the end user could use were created

**Review by rafaeltab — CHANGES_REQUESTED** [review]
More changes

**Review by rafaeltab — APPROVED** [review]
yus is good

**Review by JordyList — APPROVED** [review]
It good

**Pull request merged**